app/src/main/java/in/testpress/testpress/ui/ProductNativeGridBaseFragment.java

```diff
@@ -184,8 +183,8 @@ protected void setItemsToAdapter(){
 
     protected ArrayList<Card> initCards() {
         for (int i = 0; i < items.size(); i++) {
```

Comment:
I'm not OK with this algorithm. How does it handle pagination? Have you tested for more that 20 products?